packages/material-ui/src/ButtonBase/ButtonBase.test.js

```diff
@@ -404,34 +436,13 @@ describe('<ButtonBase />', () => {
       assert.strictEqual(onFocusVisibleSpy.callCount, 1);
       assert.strictEqual(onFocusVisibleSpy.firstCall.args.length, 1);
     });
-
-    it('should work with a functional component', () => {
-      const MyLink = React.forwardRef((props, ref) => {
-        return (
-          <a href="/foo" ref={ref} {...props}>
-            bar
-          </a>
-        );
-      });
-      const wrapper = mount(
-        <ButtonBase theme={{}} component={MyLink}>
-          Hello
-        </ButtonBase>,
-      );
-      const instance = wrapper.find('ButtonBase').instance();
-      wrapper.simulate('focus');
-      clock.tick(instance.focusVisibleCheckTime);
-    });
   });
 
-  describe('handleKeyDown()', () => {
-    describe('avoids multiple keydown presses', () => {
+  describe('event: keydown', () => {
+    // eslint-disable-next-line mocha/no-skipped-tests
+    describe.skip('avoids multiple keydown presses', () => {
```

Comment:
The behavior is not apparent to me (due to no information in the test description).

It probably wants to avoid repeated keydowns causing a ripple but I'm not sure what the benefit is. If somebody presses and holds Space and then Enter another click will be dispatched at which point we probably should create a new Ripple. Not even including that only space triggers a ripple currently.

I would skip this for now and explore in a follow-up if we can get rid of the whole "avoid repeated keydown" logic.

Reply:
> due to no information in the test description

I think that the test assertions and test actions should be what yield **most** of the information. The description is not really important.
Same as you, I don't understand what this test is about. Let's just kill it. It seems it was seeded in a 100% test coverage effort.

Reply:
> I think that the test assertions and test actions should be what yield most of the information

So in face of a cryptic test with a bad description you think tests have more information than descriptions? Why do you think these descriptions exist? Tests are an implementation detail. The description describes the API.

I'll keep it for now because I want to specifically fail it in a later PR: repeated keydown should ripple because you could press spacebar and then enter. Currently enter doesn't create a ripple but does fire a click.

Reply:
Yes, the test description is the last thing I look at when working with the tests. The shorter & generic the descriptions can be, the better. It's less surface area for an outdated description. I'm not saying the description is useless, I'm saying that the test detail is more important. It's what we should most care about. it's our insurance, the more descriptive they are, the better.

Oh, you are right, <kbd>Space</kbd> triggers the ripple, but <kbd>Enter</kbd> doesn't. I agree that something seems off.

Reply:
> outdated description

Descriptions rarely become outdated because that would mean you introduced a breaking change. This happens far less than a human reading a test.